apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_v2_SUITE.erl

```diff
@@ -0,0 +1,451 @@
+    ),
+    ok.
+
+%%------------------------------------------------------------------------------
+%% Helper fns
+%%------------------------------------------------------------------------------
+
+pulsar_connector(Config) ->
+    PulsarHost = ?config(pulsar_host, Config),
+    PulsarPort = ?config(pulsar_port, Config),
+    UseTLS = proplists:get_value(use_tls, Config, false),
+    Name = atom_to_binary(?MODULE),
+    Prefix =
+        case UseTLS of
+            true -> <<"pulsar+ssl://">>;
+            false -> <<"pulsar://">>
+        end,
+    ServerURL = iolist_to_binary([
+        Prefix,
+        PulsarHost,
+        ":",
+        integer_to_binary(PulsarPort)
+    ]),
+    Connector = #{
+        <<"connectors">> => #{
+            <<"pulsar">> => #{
+                Name => #{
+                    <<"enable">> => true,
+                    <<"ssl">> => #{
+                        <<"enable">> => UseTLS,
+                        <<"verify">> => <<"verify_none">>,
+                        <<"server_name_indication">> => <<"auto">>
+                    },
+                    <<"authentication">> => <<"none">>,
+                    <<"servers">> => ServerURL
+                }
+            }
+        }
+    },
+    parse_and_check(<<"connectors">>, emqx_connector_schema, Connector, Name).
+
+pulsar_action(Config) ->
+    Name = atom_to_binary(?MODULE),
+    Action = #{
+        <<"actions">> => #{
+            <<"pulsar">> => #{
+                Name => #{
+                    <<"connector">> => Name,
+                    <<"enable">> => true,
+                    <<"parameters">> => #{
+                        <<"retention_period">> => <<"infinity">>,
+                        <<"max_batch_bytes">> => <<"1MB">>,
+                        <<"batch_size">> => 100,
+                        <<"strategy">> => <<"random">>,
+                        <<"buffer">> => #{
+                            <<"mode">> => <<"memory">>,
+                            <<"per_partition_limit">> => <<"10MB">>,
+                            <<"segment_bytes">> => <<"5MB">>,
+                            <<"memory_overload_protection">> => true
+                        },
+                        <<"message">> => #{
+                            <<"key">> => <<"${.clientid}">>,
+                            <<"value">> => <<"${.}">>
+                        },
+                        <<"pulsar_topic">> => ?config(pulsar_topic, Config)
+                    },
+                    <<"resource_opts">> => #{
+                        <<"health_check_interval">> => <<"1s">>
+                    }
+                }
+            }
+        }
+    },
+    parse_and_check(<<"actions">>, emqx_bridge_v2_schema, Action, Name).
+
+parse_and_check(Key, Mod, Conf, Name) ->
+    ConfStr = hocon_pp:do(Conf, #{}),
+    ct:pal(ConfStr),
+    {ok, RawConf} = hocon:binary(ConfStr, #{format => map}),
+    hocon_tconf:check_plain(Mod, RawConf, #{required => false, atom_key => false}),
```

Comment:
Note: this does not check anything if we don't assert the output.  There're  `emqx_bridge_v2_testlib:parse_and_check{,_connector}` that do that.

Reply:
If the data is not fit the schema, check_plain will throw the error.

Reply:
I mean: if the raw conf is wrong and doesn't use a root that the schema knows, it basically does nothing.

For example:

```erlang
true = #{} == hocon_tconf:check_plain(emqx_connector_schema, #{<<"bridge">> => something_completely_wrong}, Opts).
```

That function from testlib will take care of that.